### DISCUSSION TITLE
Is it possible to detect support for brotli compressed JXL metadata?

### DISCUSSION THREAD

**Discussion by heirecka** [Exiv2/exiv2#3093] [Q&A] [answered]
At build time, I could check for EXV_HAVE_BROTLI but I couldn't find anything I could use for a runtime check.

**Comment by clanmills** [comment, marked as answer]
I'm not sure that I've understood the question.  Why are you interested in this.  Do you have a test image which isn't been correctly handled?

By default, brotli support is built into exiv2.  You can check to built status of exiv2 with the command `$ exiv2 --verbose --version`

```
1168 rmills@rmillsm1:~/gnu/github/exiv2/main/build $ bin/exiv2 -vV | grep brotli
library=/opt/homebrew/Cellar/brotli/1.1.0/lib/libbrotlicommon.1.1.0.dylib
library=/opt/homebrew/Cellar/brotli/1.1.0/lib/libbrotlidec.1.1.0.dylib
have_brotli=1
1169 rmills@rmillsm1:~/gnu/github/exiv2/main/build $ 
```

Is brotli being used to decompress the metadata in a JXL?  I don't know.  I retired from Exiv2 two  years ago, just as JXL surfaced.  I maintained Exiv2 0.14-v0.27.   JXL support was added to `main` and I wasn't involved.
